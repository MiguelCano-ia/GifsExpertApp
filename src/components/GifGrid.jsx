import GifItem from "./GifItem";
import useFetchGifs from "../hooks/useFetchGifs";

export const GifGrid = ({ category }) => {

  const { images, isLoading } = useFetchGifs( category );
  
  return (
    <>
      <h3>{ category }</h3>
      {
        isLoading && ( <h2>Cargando...</h2> )
      }
      
      <div className="card-grid">
        {
          images.map(( image ) => (
            <GifItem
              key={ image.id }
              { ...image }  // Esparcir los props para el hijo
            />
          ))
        }
      </div>
    </>
  )
}

export default GifGrid