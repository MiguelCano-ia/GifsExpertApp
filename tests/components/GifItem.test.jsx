import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('Pruebas en el componentes <GifItem />', () => {

  const title = 'Valorant'
  const url = 'https://valorant.com.co';

  test('deberia hacer match con el snapshot', () => {
    
    const { container } = render(
      <GifItem
        title={ title }
        url={ url } 
      />
    );
    expect( container ).toMatchSnapshot();
  });

  test('debe de mostar la imagen con el URL y el ALT indicado', () => {
    render(
      <GifItem
        title={ title }
        url={ url } 
      />
    );

    const image = screen.getByRole('img');
    expect( image.getAttribute('src') ).toBe( url );
    expect( image.getAttribute('alt') ).toBe( title );
  });
  

  test('debe de mostarar el titulo en el componente', () => {
    render(
      <GifItem
        title={ title }
        url={ url } 
      />
    );
    expect( screen.getByText( title ) ).toBeTruthy();
  });
});
