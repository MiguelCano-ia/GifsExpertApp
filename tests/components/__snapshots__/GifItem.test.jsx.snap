// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componentes <GifItem /> deberia hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Valorant"
      src="https://valorant.com.co"
    />
    <p>
      Valorant
    </p>
  </div>
</div>
`;
